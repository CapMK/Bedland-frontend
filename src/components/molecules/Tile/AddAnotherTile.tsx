import Typography from 'components/atoms/Typography/Typography';
import { useAppSelector } from 'store/hooks';
import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';

import * as Styled from './Tile.styled';
import { AddAnotherTileProps, anotherTileDefaultProps } from './Tiles.types';

const AddAnotherTile = ({ img }: AddAnotherTileProps) => {
  const { palette } = useContext(ThemeContext);
  const userRole = useAppSelector((state) => state.user.role);

  return (
    <Styled.AddAnotherTileWrapper palette={palette}>
      <Styled.BuildingImage src={img} alt={userRole === 'manager' ? 'building' : 'flat'} />
      <Styled.CardContent>
        <Typography variant='header3' color='inputGrey'>
          Add a new {userRole === 'manager' ? 'Building' : 'Flat'}
        </Typography>
      </Styled.CardContent>
    </Styled.AddAnotherTileWrapper>
  );
};

AddAnotherTile.defaultProps = anotherTileDefaultProps;

export default AddAnotherTile;
