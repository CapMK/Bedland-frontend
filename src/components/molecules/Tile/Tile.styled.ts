import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const TileWrapper = styled.div<{ palette: PaletteType }>`
  background-color: #fff;
  border-radius: 8px;
  box-shadow: ${({ palette }) => `0px 5px 8px ${palette.widgetsShadows}20`};
  color: ${({ palette }) => palette.inputGrey};
  display: flex;
  flex-direction: column;
`;

export const AddAnotherTileWrapper = styled.div<{ palette: PaletteType }>`
  align-items: center;
  background-color: #fff;
  border-radius: 8px;
  box-shadow: ${({ palette }) => `0px 5px 8px ${palette.widgetsShadows}20`};
  color: ${({ palette }) => palette.inputGrey};
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const BuildingImage = styled.img`
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
`;

export const CardContent = styled.div`
  padding: 0.5rem 1.031rem 0.938rem;
`;
