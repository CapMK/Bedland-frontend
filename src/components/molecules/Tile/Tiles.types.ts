import BuildingImg from 'assets/img/BuildingImg.png';
import AddAnother from 'assets/img/AddAnother.svg';

export type TileProps = {
  title: string;
  subtitle: string;
  id: string;
  img?: string;
};

export const tileDefaultProps: { img: string } = {
  img: BuildingImg,
};

export type AddAnotherTileProps = {
  img?: string;
};

export const anotherTileDefaultProps: { img: string } = {
  img: AddAnother,
};
