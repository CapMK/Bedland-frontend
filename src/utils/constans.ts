export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
