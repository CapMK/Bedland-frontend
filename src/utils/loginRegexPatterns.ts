import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
