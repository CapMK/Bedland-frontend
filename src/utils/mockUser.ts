import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
