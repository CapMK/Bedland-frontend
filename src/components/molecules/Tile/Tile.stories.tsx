import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';

import Tile from './Tile';

const meta = {
  title: 'Molecules/Tile',
  component: Tile,
  tags: ['autodocs'],
  argTypes: {
    title: {
      control: 'text',
      description: 'Title of the tile',
    },
    subtitle: {
      control: 'text',
      description: 'Subtitle of the tile',
    },
    img: {
      table: {
        disable: true,
      },
      id: {
        control: 'text',
        description: 'Tile number',
      },
    },
  },
  parameters: {
    backgrounds: { default: 'light' },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Tile>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Main: Story = {
  args: {
    subtitle: 'Cracow, Poland',
    title: 'Berry Street 27',
    id: '1',
  },
};
