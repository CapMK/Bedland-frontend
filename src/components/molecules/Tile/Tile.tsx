import Button from 'components/atoms/Button/Button';
import Typography from 'components/atoms/Typography/Typography';
import { useAppSelector } from 'store/hooks';
import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import routes from 'App/routing/routes';
import { NavLink } from 'react-router-dom';

import * as Styled from './Tile.styled';
import { TileProps, tileDefaultProps } from './Tiles.types';

const Tile = ({ title, subtitle, img, id }: TileProps) => {
  const { palette } = useContext(ThemeContext);
  const userRole = useAppSelector((state) => state.user.role);

  return (
    <Styled.TileWrapper palette={palette} role='listitem'>
      <Styled.BuildingImage src={img} alt={userRole === 'manager' ? 'building' : 'flat'} />
      <Styled.CardContent>
        <Typography variant='header3' color='inputGrey'>
          {title}
        </Typography>
        <Typography variant='paragraph' color='inputGrey'>
          {subtitle}
        </Typography>
        <Button wrapper marginTop={1}>
          <NavLink to={routes.buildingDetails.replace(':id', id)}>
            <Button variant='primary' color='blue'>
              Open details
            </Button>
          </NavLink>
        </Button>
      </Styled.CardContent>
    </Styled.TileWrapper>
  );
};

Tile.defaultProps = tileDefaultProps;

export default Tile;
