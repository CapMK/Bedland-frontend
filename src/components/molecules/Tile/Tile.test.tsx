import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import BuildingImage from 'assets/img/BuildingImg.png';
import { loggedInManagerUser, loggedInResidentUser } from 'utils/mockUser';
import Tile from './Tile';

describe('Tile of display list', () => {
  afterEach(cleanup);
  it('renders correctly for resident', () => {
    const page = renderWithProviders(
      <Tile title='4293' subtitle='Berry Street 27' img={BuildingImage} id='1' />,
      {
        preloadedState: {
          user: loggedInResidentUser,
        },
      },
    );
    expect(page).toMatchSnapshot();
  });

  it('renders correctly for manager', () => {
    const page = renderWithProviders(
      <Tile title='4293' subtitle='Berry Street 27' img={BuildingImage} id='1' />,
      {
        preloadedState: {
          user: loggedInManagerUser,
        },
      },
    );
    expect(page).toMatchSnapshot();
  });
});
