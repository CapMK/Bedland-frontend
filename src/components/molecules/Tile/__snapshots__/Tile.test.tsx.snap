// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tile of display list > renders correctly for manager 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  margin-top: 1rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c0 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c2 {
  padding: 0.5rem 1.031rem 0.938rem;
}

<body>
    <div>
      <div
        class="c0"
        role="listitem"
      >
        <img
          alt="building"
          class="c1"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c2"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            4293
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Berry Street 27
          </p>
          <div
            class="c3"
          >
            <a
              class=""
              href="/buildings/1"
            >
              <button
                class="c4"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c3 {
  margin-top: 1rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c0 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c2 {
  padding: 0.5rem 1.031rem 0.938rem;
}

<div>
    <div
      class="c0"
      role="listitem"
    >
      <img
        alt="building"
        class="c1"
        src="/src/assets/img/BuildingImg.png"
      />
      <div
        class="c2"
      >
        <h3
          style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
        >
          4293
        </h3>
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
        >
          Berry Street 27
        </p>
        <div
          class="c3"
        >
          <a
            class=""
            href="/buildings/1"
          >
            <button
              class="c4"
            >
              Open details
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Tile of display list > renders correctly for resident 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  margin-top: 1rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c0 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c2 {
  padding: 0.5rem 1.031rem 0.938rem;
}

<body>
    <div>
      <div
        class="c0"
        role="listitem"
      >
        <img
          alt="flat"
          class="c1"
          src="/src/assets/img/BuildingImg.png"
        />
        <div
          class="c2"
        >
          <h3
            style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
          >
            4293
          </h3>
          <p
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            Berry Street 27
          </p>
          <div
            class="c3"
          >
            <a
              class=""
              href="/buildings/1"
            >
              <button
                class="c4"
              >
                Open details
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c3 {
  margin-top: 1rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c0 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c2 {
  padding: 0.5rem 1.031rem 0.938rem;
}

<div>
    <div
      class="c0"
      role="listitem"
    >
      <img
        alt="flat"
        class="c1"
        src="/src/assets/img/BuildingImg.png"
      />
      <div
        class="c2"
      >
        <h3
          style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
        >
          4293
        </h3>
        <p
          style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
        >
          Berry Street 27
        </p>
        <div
          class="c3"
        >
          <a
            class=""
            href="/buildings/1"
          >
            <button
              class="c4"
            >
              Open details
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
